traitlets/config/configurable.py

```diff
@@ -187,6 +187,10 @@ def _config_changed(self, change):
 
     def update_config(self, config):
         """Update config and load the new values"""
+        # make a copy of our config prior to loading to protect IPython,
```

Comment:
I've updated the comment to make it clear that IPython is **not** making a proper, supported assumption. I left it out of the docstring, because I don't want to commit to the behavior. This is specifically reintroducing undesirable behavior for backward compatibility in support of IPython in particular due to an accidental assumption in IPython of unspecified, undocumented behavior.